src/modules/PulseTransfer/PulseTransferModule.cpp

```diff
@@ -178,12 +178,31 @@ void PulseTransferModule::run(unsigned int event_num) {
             auto pulse_graph = new TGraph(static_cast<int>(pulse_vec.size()), &time[0], &pulse_vec[0]);
             pulse_graph->GetXaxis()->SetTitle("t [ns]");
             pulse_graph->GetYaxis()->SetTitle("Q_{ind} [e]");
-            pulse_graph->SetTitle(("Induced charge in pixel (" + std::to_string(index.x()) + "," +
+            pulse_graph->SetTitle(("Induced charge carriers per unit step time in pixel (" + std::to_string(index.x()) + "," +
                                    std::to_string(index.y()) +
-                                   "), Q_{tot} = " + std::to_string(pixel_index_pulse.second.getCharge()) + " e")
+                                   "), Q_{tot} = " + std::to_string(pixel_index_pulse.second.getCharge()) + " e (" + std::to_string(pixel_index_pulse.second.getCharge() * (-1.602e-4)) + " fC)")
                                       .c_str());
             getROOTDirectory()->WriteTObject(pulse_graph, name.c_str());
 
+            std::vector<double> current_vec;
+            double current = 0;
+            for(const auto& c_bin : pulse_vec) {
```

Comment:
Here, you could use

```cpp
auto current_vec = pulse_vec;
std::for_each(current_vec.begin(), current_vec.end(), [&step = time.front()](double &bin){ bin = bin/step * 1.602e-4; });
```

See: http://en.cppreference.com/w/cpp/algorithm/for_each

Reply:
I tried adding replacing my code with this, It does not work. No plots for current are being generated anymore.

Reply:
No worries, that's something we can do later. Thanks a lot for the contribution!